# Research Question 1:Can we predict the box office success of a movie based on its  rating, budget, cast, and other features.   What factors are most strongly correlated with box office success, and how accurate are the predictions

some factors that have been found to be strongly correlated with box office success include:

Budget: Generally, higher-budget movies tend to have higher box office revenues. However, this correlation may not hold for all movies, and there may be a point of diminishing returns for very high-budget movies.

Cast and crew: Having well-known and popular actors, directors, and writers can attract audiences and generate buzz around a movie, which can translate into higher box office revenues.

Pre-release marketing: A strong marketing campaign can generate excitement and anticipation for a movie, leading to higher box office revenues. This includes advertising, press coverage, social media, and promotional events.

# Pre-release marketing and Box office

In [1]:
import pandas as pd
import statsmodels.api as sm
import plotly.express as px
import plotly.graph_objs as go
from plotly.subplots import make_subplots
from sklearn.linear_model import LinearRegression


C:\Users\malsh\anaconda3\lib\site-packages\numpy\_distributor_init.py:30: UserWarning: loaded more than 1 DLL from .libs:
C:\Users\malsh\anaconda3\lib\site-packages\numpy\.libs\libopenblas.FB5AE2TYXYH2IJRDKGDGQ3XBKLKTF43H.gfortran-win_amd64.dll
C:\Users\malsh\anaconda3\lib\site-packages\numpy\.libs\libopenblas64__v0.3.21-gcc_10_3_0.dll
  warnings.warn("loaded more than 1 DLL from .libs:"


In [2]:
df_movies_marketing = pd.read_csv('Finalmovie_dataset_fin1.csv')

In [3]:
df_movies_marketing.shape

(44872, 15)

In [4]:
df_movies_marketing.head()

budget popularity                               production_companies  \
0       0   2.561161        [{'name': 'Walt Disney Pictures', 'id': 2}]   
1       0  12.140733  [{'name': 'Universal Pictures', 'id': 33}, {'n...   
2       0   2.228434                 [{'name': 'BBC Films', 'id': 288}]   
3       0   1.100915  [{'name': 'Ministère des Affaires Étrangères',...   
4       0   0.745542    [{'name': 'Iwerks Entertainment', 'id': 70801}]   

                                             country  Box_office  runtime  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...           0     97.0   
1  [{'iso_3166_1': 'US', 'name': 'United States o...    11348324     78.0   
2  [{'iso_3166_1': 'GB', 'name': 'United Kingdom'...           0    104.0   
3  [{'iso_3166_1': 'CN', 'name': 'China'}, {'iso_...           0    108.0   
4  [{'iso_3166_1': 'FR', 'name': 'France'}, {'iso...           0     50.0   

               name    genre released  imdb Rating director writer Cast  \
0      Tom and Huck   Action      NaN          NaN      NaN    NaN  NaN   
1             Balto   Family      NaN          NaN      NaN    NaN  NaN   
2        Persuasion    Drama      NaN          NaN      NaN    NaN  NaN   
3    Shanghai Triad    Drama      NaN          NaN      NaN    NaN  NaN   
4  Wings of Courage  Romance      NaN          NaN      NaN    NaN  NaN   

     profit  year_only  
0         0     1995.0  
1  11348324     1995.0  
2         0     1995.0  
3         0     1995.0  
4         0     1996.0

# yotube Movie trailers

In [6]:
df_movies_yotube = pd.read_csv('Movie_Trailers_ready_2.csv')

In [7]:
df_movies_yotube.head()

Movie Name (Trailer)     Views    Likes
0              Dawg Fight     99901      157
1                   Joker  99868787  1497107
2  Brothers McMullen, The      9923       24
3               Insidious   9913503    43902
4                 Ronaldo    990775     4802

In [8]:
df_movies_yotube.shape

(1124, 3)

In [9]:
df_movies_yotube.describe

<bound method NDFrame.describe of                 Movie Name (Trailer)     Views    Likes
0                         Dawg Fight     99901      157
1                              Joker  99868787  1497107
2             Brothers McMullen, The      9923       24
3                          Insidious   9913503    43902
4                            Ronaldo    990775     4802
...                              ...       ...      ...
1119          The Kids Are All Right   1000539     2641
1120  Peanuts - Die Bank zahlt alles         1        0
1121                Shadows (Cienie)         0        0
1122                           Gospa         0        0
1123                  Superweib, Das         0        0

[1124 rows x 3 columns]>

In [10]:
df_movies_yotube = df_movies_yotube.rename(columns={'Movie Name (Trailer)': 'name'})

In [11]:
df_movies_yotube

name     Views    Likes
0                         Dawg Fight     99901      157
1                              Joker  99868787  1497107
2             Brothers McMullen, The      9923       24
3                          Insidious   9913503    43902
4                            Ronaldo    990775     4802
...                              ...       ...      ...
1119          The Kids Are All Right   1000539     2641
1120  Peanuts - Die Bank zahlt alles         1        0
1121                Shadows (Cienie)         0        0
1122                           Gospa         0        0
1123                  Superweib, Das         0        0

[1124 rows x 3 columns]

# Merging the Two Dataframes 

In [20]:
df_merged = pd.merge(df_movies_marketing, df_movies_yotube, on='name')


In [21]:
# Define the feature matrix and target variable
X = df_merged[['Likes','Views']]
y = df_merged['Box_office']


In [22]:
X = sm.add_constant(X)


In [23]:
ols_model = sm.OLS(y, X).fit()


In [24]:
print(ols_model.summary())


                            OLS Regression Results                            
Dep. Variable:             Box_office   R-squared:                       0.346
Model:                            OLS   Adj. R-squared:                  0.345
Method:                 Least Squares   F-statistic:                     261.3
Date:                Thu, 18 May 2023   Prob (F-statistic):           8.04e-92
Time:                        14:10:28   Log-Likelihood:                -20473.
No. Observations:                 991   AIC:                         4.095e+04
Df Residuals:                     988   BIC:                         4.097e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       5.933e+07   8.92e+06      6.654      0.0

This OLS regression analysis investigates the relationship between the dependent variable "Box_office" and the independent variables "Likes" and "Views." The model's R-squared value is 0.346, indicating that approximately 34.6% of the variation in box office can be explained by the two predictors. The F-statistic is 261.3, with a p-value of 8.04e-92, suggesting the model's overall statistical significance.

The coefficient for the constant term is 5.933e+07, representing the expected box office value when both "Likes" and "Views" are zero. The coefficient for "Likes" is -490.4960, indicating that, on average, each additional like is associated with a decrease of $490.50 in box office revenue. The coefficient for "Views" is 13.9830, suggesting that, on average, for each additional view, the box office is expected to increase by approximately $13.98.

The standard errors, t-statistics, and p-values assess the statistical significance of the coefficients. Both "Likes" and "Views" have small p-values (<0.05), indicating their statistical significance.

Other statistical measures such as AIC, BIC, omnibus test, Jarque-Bera test, skewness, and kurtosis provide further insights into the model's performance and the assumption of normality for the residuals.

In summary, this analysis indicates that the number of likes and views have a statistically significant relationship with box office revenue, with more views contributing positively and more likes potentially having a negative impact on box office performance.

In [25]:
df_merged.head()

budget popularity                               production_companies  \
0       0   0.046196                                                 []   
1       0   6.791697  [{'name': 'Skyhook Productions', 'id': 59566},...   
2       0   3.288353  [{'name': 'Samba P.C.', 'id': 77451}, {'name':...   
3       0    1.57501  [{'name': 'Lightning Entertainment', 'id': 325...   
4       0   0.202054                                                 []   

                                             country  Box_office  runtime  \
0            [{'iso_3166_1': 'IT', 'name': 'Italy'}]           0     92.0   
1  [{'iso_3166_1': 'US', 'name': 'United States o...           0     88.0   
2            [{'iso_3166_1': 'ES', 'name': 'Spain'}]           0     86.0   
3  [{'iso_3166_1': 'US', 'name': 'United States o...           0     91.0   
4                                                 []           0     93.0   

             name     genre released  imdb Rating director writer Cast  \
0      The Escort  Thriller      NaN          NaN      NaN    NaN  NaN   
1      The Escort     Drama      NaN          NaN      NaN    NaN  NaN   
2         Anguish     Crime      NaN          NaN      NaN    NaN  NaN   
3         Anguish    Horror      NaN          NaN      NaN    NaN  NaN   
4  Boy Meets Girl     Drama      NaN          NaN      NaN    NaN  NaN   

   profit  year_only    Views  Likes  
0       0     1993.0  2631629   8354  
1       0     2015.0  2631629   8354  
2       0     1987.0   289724    605  
3       0     2015.0   289724    605  
4       0     1994.0  1146945   5308

In [26]:
import plotly.express as px
import plotly.graph_objects as go
from sklearn.linear_model import LinearRegression

df_merged = pd.merge(df_movies_marketing, df_movies_yotube, on='name')

# First scatter plot
X_views = df_merged[['Views', 'Likes']]
y_views = df_merged['Box_office']
reg_views = LinearRegression().fit(X_views, y_views)
r_squared_views = reg_views.score(X_views, y_views)

fig_views = px.scatter(df_merged, x='Views', y='Box_office', hover_data=['name', 'Likes'])
fig_views.update_traces(marker=dict(color='red'))
fig_views.update_layout(
    title=f'Box Office Revenue vs. Number of Views for Movie Trailers (R-squared: {r_squared_views:.2f})',
    xaxis_title='Number of Views',
    yaxis_title='Box Office Revenue',
    font=dict(size=14)
)

x_range_views = [df_merged['Views'].min(), df_merged['Views'].max()]
y_range_views = [reg_views.predict([[x_range_views[0], 0]])[0], reg_views.predict([[x_range_views[1], 0]])[0]]
fig_views.add_trace(go.Scatter(x=x_range_views, y=y_range_views, mode='lines', name='Linear Regression', line=dict(color='black', width=2)))
fig_views.update_layout(template="simple_white")
fig_views.update_layout(yaxis_range=[0, 500000000])
fig_views.update_layout(xaxis_range=[0, 50000000])
fig_views.show()

# Second scatter plot
X_likes = df_merged[['Views', 'Likes']]
y_likes = df_merged['Box_office']
reg_likes = LinearRegression().fit(X_likes, y_likes)
r_squared_likes = reg_likes.score(X_likes, y_likes)

fig_likes = px.scatter(df_merged, x='Likes', y='Box_office', hover_data=['name', 'Views'])
fig_likes.update_traces(marker=dict(color='blue'))
fig_likes.update_layout(
    title=f'Box Office Revenue vs. Number of Likes for Movie Trailers (R-squared: {r_squared_likes:.2f})',
    xaxis_title='Number of Likes',
    yaxis_title='Box Office Revenue',
    font=dict(size=14)
)

x_range_likes = [df_merged['Likes'].min(), df_merged['Likes'].max()]
y_range_likes = [reg_likes.predict([[0, x_range_likes[0]]])[0], reg_likes.predict([[0, x_range_likes[1]]])[0]]
fig_likes.add_trace(go.Scatter(x=x_range_likes, y=y_range_likes, mode='lines', name='Linear Regression', line=dict(color='black', width=2)))
fig_likes.update_layout(template="simple_white")
fig_likes.update_layout(yaxis_range=[0, 500000000])
fig_likes.update_layout(xaxis_range=[0, 250000])
fig_likes.show()


C:\Users\malsh\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning:

X does not have valid feature names, but LinearRegression was fitted with feature names

C:\Users\malsh\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning:

X does not have valid feature names, but LinearRegression was fitted with feature names



C:\Users\malsh\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning:

X does not have valid feature names, but LinearRegression was fitted with feature names

C:\Users\malsh\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning:

X does not have valid feature names, but LinearRegression was fitted with feature names

